File: Java/ContainsDuplicate/Solution2.java
import java.util.HashMap;
import java.util.Map;
/**
 *(11ms)
 * 方法二：
 * 这个方法实际上与其他两种方法有相似之处，比方法一好在当发现存在重复元素直接返回，而不用将所有元素加入到map中
 * 而比方法三好在代码简洁，但是HashMap的存取会消耗一定的时间。
 */
public class Solution2 {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for(Integer num : nums){
            if(map.get(num) != null) {
                return true;
            }
            map.put(num, num);
        }
        return false;
    }
}

File: Python/ContainsDuplicate/Solution2.py
#(45ms)这个解法很优美，只有一行代码，与我在java中的第一个解法的想法是一样的
#将nums转换成set，如果set的长度小于nums的长度，则一定存在重复的元素
#与java不同的是，这种做法比使用dict更快。。。与java正好相反
class Solution(object):
    def containsDuplicate(self, nums):
		"""
        :type nums: List[int]
        :rtype: bool
        """
		return len(set(nums)) < len(nums)

test = Solution()
print test.containsDuplicate([1, 0, -1, 1])